src/modules/airspeed_selector/AirspeedValidator.cpp

```diff
@@ -149,6 +149,11 @@ AirspeedValidator::check_airspeed_innovation(uint64_t time_now, float estimator_
 		_innovations_check_failed = false;
 		_time_last_tas_pass = time_now;
 
+	} else if (_tas_innov_integ_threshold <= 0.f) {
```

Comment:
Agree, the current logic is suboptimal in case of either bad mag/velocity data or a disabled integrator.

How about combining that condition (_tas_innov_integ_threshold <= 0.f) with the previous one? (if (!_in_fixed_wing_flight ..))